files/default/neutron-ha-tool.py

```diff
@@ -263,57 +265,45 @@ def l3_agent_migrate(qclient, noop=False, now=False):
 
     """
 
-    migration_count = 0
     agent_list = list_agents(qclient)
     agent_dead_list = agent_dead_id_list(agent_list, 'L3 agent')
     agent_alive_list = agent_alive_id_list(agent_list, 'L3 agent')
     LOG.info("There are %s offline L3 agents and %s online L3 agents",
              len(agent_dead_list), len(agent_alive_list))
 
-    if len(agent_dead_list) > 0:
-        if len(agent_alive_list) < 1:
-            LOG.exception("There are no l3 agents alive to migrate "
-                          "routers onto")
-
-        timeout = 0
-        if not now:
-            while timeout < TAKEOVER_DELAY:
-
-                agent_list_new = list_agents(qclient)
-                agent_dead_list_new = agent_dead_id_list(agent_list_new,
-                                                         'L3 agent')
-                if len(agent_dead_list_new) < len(agent_dead_list):
-                    LOG.info("Skipping router failover since an agent came "
-                             "online while ensuring agents offline for %s "
-                             "seconds", TAKEOVER_DELAY)
-                    sys.exit(0)
-
-                LOG.info("Agent found offline for seconds=%s but waiting "
-                         "seconds=%s before migration",
-                         timeout, TAKEOVER_DELAY)
-                timeout += 1
-                time.sleep(1)
-
-        for agent_id in agent_dead_list:
-            LOG.info("Querying agent_id=%s for routers to migrate", agent_id)
-            router_id_list = list_routers_on_l3_agent(qclient, agent_id)
+    if len(agent_dead_list) == 0:
+        return
 
-            for router_id in router_id_list:
+    if len(agent_alive_list) < 1:
+        LOG.error("There are no l3 agents alive to migrate routers onto - "
+                  "aborting!")
+        sys.exit(1)
```

Comment:
Oh yeah, for some reason I was thinking the l3 agent handled dhcp too.

Whilst your concern is technically valid, it's kind of irrelevant because [dhcp rebalancing is done in a separate invocation of this script](https://github.com/openstack/openstack-resource-agents/blob/master/ocf/neutron-ha-tool#L218).  The only case which would matter is if someone invoked `neutron-ha-tool --l3-agent-migrate --replicate-dhcp` manually, which seems unlikely ... and even then, I wonder how useful working dhcp agents really are if there are no l3 agents.  But I can change it if you really think it's worth it.